netebpfext/net_ebpf_ext.c

```diff
@@ -632,6 +634,46 @@ _net_ebpf_ext_provider_client_attach_callback(
     return EBPF_SUCCESS;
 }
 
+static _Function_class_(KDEFERRED_ROUTINE) _IRQL_requires_max_(DISPATCH_LEVEL) _IRQL_requires_min_(DISPATCH_LEVEL)
+    _IRQL_requires_(DISPATCH_LEVEL) _IRQL_requires_same_ VOID _net_ebpf_ext_rundown(
+        _In_ struct _KDPC* Dpc,
+        _In_opt_ PVOID DeferredContext,
+        _In_opt_ PVOID SystemArgument1,
+        _In_opt_ PVOID SystemArgument2)
+{
+    net_ebpf_ext_hook_provider_registration_t* registration =
+        (net_ebpf_ext_hook_provider_registration_t*)DeferredContext;
+
+    UNREFERENCED_PARAMETER(Dpc);
+    UNREFERENCED_PARAMETER(SystemArgument1);
+    UNREFERENCED_PARAMETER(SystemArgument2);
+    if (registration)
+        KeSetEvent(&registration->rundown_wait, 0, FALSE);
+}
+
+static void
+_net_ebpf_ext_init_rundown(net_ebpf_ext_hook_provider_registration_t* registration)
+{
+    KeInitializeEvent(&(registration->rundown_wait), NotificationEvent, FALSE);
+    KeInitializeDpc(&(registration->rundown_dpc), _net_ebpf_ext_rundown, registration);
+}
+
+static void
+_net_ebpf_ext_wait_for_rundown(net_ebpf_ext_hook_provider_registration_t* registration)
+{
+    // Queue a DPC to each CPU and wait for it to run.
+    // After it has run on each CPU we can be sure that no
+    // DPC is busy processing a hook.
+    ULONG maximum_processor = KeQueryMaximumProcessorCount();
+    ULONG processor;
```

Comment:
can standard types like uint32_t be used here instead of ULONG or does that give a compiler error?